### PR TITLE
Two new methods added to get an array of values for given key search

### PR DESCRIPTION
**Pull request by razcakappa**
Also added the functionality to get the array of matching keys for a given phrase

### CODE REVIEW

**Comment by razcakappa** [comment]
@royriojas Two functionalities added to retrieve an array of values by a search phrase for keys. And also get an array of matched keys.

File: cache.js [hunk 1]
```diff
@@ -80,6 +80,41 @@ var cache = {
     this._visited[ key ] = true;
     return this._persisted[ key ];
   },
+  /**
+   * Return the values set for the matching keys
+   * @method getKeysByMatch
+   * @param filter {String} the search string of the keys to retrieve
+   * @returns {values, ...} the values from the matched key set
+   */
+  getKeysByMatch: function ( filter ) {
```
Comment (royriojas) [inline cache.js]: Interesting, what would be the use case for this function?

Also`getKeyValuesByMatch` might be a better name I think and the other function can be `getKeysByMatch`

Comment (razcakappa) [reply cache.js]: This returns an array of cached objects(whole) or just the keys by looking up with a partial phrase of a key.

Real world example:
I had to store access tokens with user ids with some custom value as 
```
_____________________________________
|    key                              |     value                  |
|____________________|________________|
|    'userID:accessToken'  |     {custom: vlaue}  |
|____________________|________________|
```

And I had to retrieve all objects which belong to single userID
Also I had to retrieve the object which matches with a given accessToken

I agree with the name change of the function

Comment (razcakappa) [reply cache.js]: Also I've done the function name change

File: cache.js [hunk 2]
```diff
@@ -80,6 +80,41 @@ var cache = {
...
+      keys = [];
+    for (key in this._persisted) {
+      if ( this._persisted.hasOwnProperty( key ) && key.includes( filter ) ) {
+        keys.push( this._persisted[ key ] );
+        this._visited[ key ] = true;
+
+      }
+    }
+    return keys;
+  },
```
Comment (razcakappa) [inline cache.js]: Also I've done the function name change

**Comment by jaredwray** [comment]
@razcakappa - we are closing this down as it will not work with the updated version we are working on.

**Pull request closed without merging**